internal/provider/job_resource.go

```diff
@@ -0,0 +1,334 @@
+	if req_data != nil {
+		http_code, body, err = r.client.doRequest("POST", post_url, req_data)
+	} else {
+		http_code, body, err = r.client.doRequest("POST", post_url, nil)
+	}
+
+	if err != nil {
+		return err
+	}
+	if http_code != http.StatusCreated {
+		return fmt.Errorf("the server returned status code %d while attempting to create Job", http_code)
+	}
+	err = data.ParseHttpResponse(body)
+	if err != nil {
+		return fmt.Errorf("error while parsing the json response: " + err.Error())
+	}
+	return nil
+}
+
+func (r JobResource) ReadJob(data JobResourceModelInterface) error {
+	// Read existing Job
+	jobURL := data.GetURL()
+	if len(jobURL) > 0 {
+		http_code, body, err := r.client.doRequest("GET", jobURL, nil)
+		if err != nil {
+			return err
+		}
+
+		if http_code != http.StatusOK {
+			return fmt.Errorf("the server returned status code %d while attempting to Get from URL %s", http_code, jobURL)
+		}
+
+		err = data.ParseHttpResponse(body)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r JobResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data jobResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	err := r.ReadJob(&data)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Read error",
+			err.Error(),
+		)
+		return
+	}
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r JobResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data jobResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := r.CreateJob(&data); err != nil {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Creation error",
+			err.Error(),
+		)
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r JobResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+}
+
+func (r JobResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data jobResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	// Create new Job from job template
+	if err := r.CreateJob(&data); err != nil {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Update error",
+			err.Error(),
+		)
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *JobResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("job_template_id"), req, resp)
```

Comment:
I don't think this will work because `ImportStatePassthroughID` requires that the state attribute accept a string but `"job_template_id"` is int64. Do we need to support state import for a job resource?